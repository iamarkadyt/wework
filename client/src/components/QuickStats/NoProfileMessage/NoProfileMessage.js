import React, { Fragment } from 'react'
import Field from '../../Field/Field'

const NoProfileMessage = ({ history }) => (
    <Fragment>
        <h2>You don't yet have a profile, create one now!</h2>
        <Field 
            type="button"
            label="Create"
            onClick={() => {
                history.push('/profile/create-profile')
            }} />
    </Fragment>
)

export default NoProfileMessage
