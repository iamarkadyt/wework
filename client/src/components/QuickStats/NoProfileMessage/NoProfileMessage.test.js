import React from 'react'
import { shallow } from 'enzyme'
import NoProfileMessage from './NoProfileMessage'
import cloneDeep from 'lodash.clonedeep'

let mountedComponent, props

const getMockProps = () => {
  return cloneDeep({
    history: {
      push: jest.fn()
    }
  })
}

const comp = () => {
  if (!mountedComponent) {
    mountedComponent = shallow(<NoProfileMessage {...props} />)
  }
  return mountedComponent
}

describe('NoProfileMessage', () => {
  beforeEach(() => {
    mountedComponent = undefined
    props = getMockProps()
  })

  it('matches snapshot', () => {
    expect(comp()).toMatchSnapshot()
  })

  describe('Create button', () => {
    it('receives correct function into onClick prop', () => {
      comp().find('field').simulate('click')
      expect(props.history.push).toHaveBeenCalled()
    })

    it('receives correct function into onClick prop, and uses it correctly', () => {
      comp().find('field').simulate('click')
      expect(props.history.push).toHaveBeenCalledWith('/profile/create-profile')
    })
  })
})
