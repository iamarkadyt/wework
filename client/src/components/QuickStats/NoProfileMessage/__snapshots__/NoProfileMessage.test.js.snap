// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NoProfileMessage matches snapshot 1`] = `
<Fragment>
  <h2>
    You don't yet have a profile, create one now!
  </h2>
  <field
    label="Create"
    onClick={[Function]}
    type="button"
  />
</Fragment>
`;
